### ISSUE TITLE
dnvm setup should set DNX_HOME machine-wide on Windows

### ISSUE THREAD

**Issue by anurse** [aspnet/dnvm#439]
When we install DNVM on Windows, we should set the `DNX_HOME` environment variable to the appropriate value (including global install dirs). This should be in the `dnvm-setup` function, and should run only if elevated.

The MSI should (already?) set the same environment variable so this would only be for scenarios where DNVM is installed via the "copy-pasta" script.

/cc @marcind @BrennanConroy @glennc

**Comment by marcind** [comment]
Make sure the MSI is available via Web Platform Installer.

**Comment by anurse** [comment]
@danroth27 @DamianEdwards Are there plans to integrate our MSIs into WebPI?

**Comment by glennc** [comment]
I haven't talked about adding it to WebPI with anyone yet, but it should be possible. I take it you like WebPI and want to use that to install all of our stuff?

@anurse this is just to prep the machine for wherever we may drop DNXs in the future?

**Comment by marcind** [comment]
@glennc I'm looking for any method that allows me to install dependencies onto a target host in a 'one-liner' declarative, idempotent fashion (i.e. I can automate this). WebPI and Chocolatey fit that bill as I can just choose the product and version and they take care of the rest (including no-op if it's already installed). No need to look up download links, manage MSIs, etc. So if you're going to have an MSI, make sure to add it to those registries.

Though honestly, I've been quite happy with the capabilities of DNVM so far (esp. once the machine-wide install story gets completed).

**Comment by anurse** [comment]
We've also had people ask about PowerShell Package Management/Desired State Configuration support which is compatible with MSIs.

In general, I think a major RC-level thing will be refining our acquisition story so feedback here is welcome :). In this case, `dnvm` could have (and basically does already) it's own idempotent "machine setup" script so that all you have to do is bring `dnvm.ps1` to the box, run `setup` and maybe `update-self` to get the true latest (if you didn't directly download the latest) and you should be set.

**Comment by marcind** [comment]
Since we're talking about covering all the bases here, having this available via `apt-get` on Ubuntu would also be great :)

**Comment by DamianEdwards** [comment]
@marcind we're working on that right now. We plan to have the following acquisition experiences ready for RC in November (some may come online earlier):
- Scripted install for all platforms (curl a script and run it to get DNVM)
- DNVM for all platforms (including support for global/shared installs)
- EXE installer for Windows (with CLI options to specify what gets installed)
- PKG installer for OSX
- apt-get feed for Ubuntu and related distros
- TAR download for Linux distros

**Comment by dougbu** [comment]
Looking at #473, I see we need to revisit the approach for this issue.
> The MSI should (already?) set the same environment variable

Is no longer true. Nothing sets `$env:DNX_HOME`, `$env:DNX_GLOBAL_HOME` or `$env:DNX_USER_HOME`. Their use is reserved for overrides since the scripts et cetera know how to calculate correct default values for every user.

That said machine-wide settings may be useful for Windows services or machine-wide installations in non-default locations. Perhaps this option needs a separate `dnvm` command or off-by-default option for `dnvm setup`...

**Comment by anurse** [comment]
@dougbu @cesarbs and I chatted and there are some other issues in this. We'll try to resync on this next week (maybe try to track down @davidfowl). A couple things noted:

* DNVM is currently inconsistent with dealing with `DNX_USER_HOME`, `DNX_GLOBAL_HOME` and `DNX_HOME`. When `DNX_USER_HOME`/`DNX_GLOBAL_HOME` is missing, we try to recreate them from `DNX_HOME` by picking the one in Program Files as the global home. This is wrong now
* Setting `DNX_HOME` at a machine level is really tricky because Windows **really** likes expanding environment variables :). It's possible, but a little difficult
* Maybe we shouldn't bother setting it. We continue to read it so hosts like Antares can **override** the default values. MSIs/DEBs/RPMs/PKGs/etc. will still install stuff into default paths and scripts can search those paths as well if `DNX_HOME` isn't provided.

I think I covered most things, but @dougbu or @cesarbs can expand on this if not :)
